components/policy/brave_browser_policy_provider.h

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_POLICY_BRAVE_BROWSER_POLICY_PROVIDER_H_
+#define BRAVE_COMPONENTS_POLICY_BRAVE_BROWSER_POLICY_PROVIDER_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "components/policy/core/common/configuration_policy_provider.h"
+
+class PrefService;
+
+namespace brave_policy {
+
+// Browser-level policy provider for global scope preferences.
+// This provider handles policies that affect local state (global) preferences
+// and is registered with the BrowserPolicyConnector for machine-wide policy
+// management.
+class BraveBrowserPolicyProvider : public policy::ConfigurationPolicyProvider {
```

Comment:
Should we consider an Origin-specific name i.e. `OriginBrowserPolicyProvider`? Future policy types unrelated to Origin could have their own policy provider.